ob1k-cache/src/test/java/com/outbrain/ob1k/cache/LocalAsyncCacheTest.java

```diff
@@ -139,95 +139,11 @@ public void testBadLoader() {
     assertComposableFutureError(null,NullPointerException.class, cache.getAsync("null"));
   }
 
-  public static class Box {
-    public final int number;
-
-    public Box(final int number) {
-      this.number = number;
-    }
-  }
-
-  public static class BoxUpdater extends Thread {
-    private final TypedCache<String, Box> cache;
-    private final String key;
-    private final int iterations;
-    private volatile int successCounter;
-
-    public BoxUpdater(final TypedCache<String, Box> cache, final String key, final int iterations) {
-      this.cache = cache;
-      this.key = key;
-      this.iterations = iterations;
-    }
-
-    @Override
-    public void run() {
-      int success = 0;
-      int failure = 0;
-      for (int i =0; i < iterations; i++) {
-        try {
-          final boolean res = cache.setAsync(key, new EntryMapper<String, Box>() {
-            @Override
-            public Box map(final String key, final Box value) {
-              return new Box(value.number + 1);
-            }
-          }, 100).get();
-
-          if (res) {
-            success++;
-          } else {
-            failure++;
-          }
-
-        } catch (final Exception e) {
-          e.printStackTrace();
-        }
-
-      }
-      successCounter = success;
-    }
-
-    public int getSuccessCounter() {
-      return successCounter;
-    }
-  }
-
-  @Test
-  public void testCasWithMapper() throws Exception {
```

Comment:
instead of removing this test and testing only one impl, add an abstract test base class, and have the two implementations tests extend it.